### PR TITLE
Update pins for boost and boost_cpp

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3402

### CODE REVIEW

File: recipe/migrations/boost1780.yaml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 __migrator:
   build_number: 1
   kind: version
-  migration_number: 1
+  migration_number: 2
 boost:
 - 1.78.0
+boost_cpp
+- 1.78.0
 migrator_ts: 1649717600.9611828
```
Comment (hmaarrfk) [inline recipe/migrations/boost1780.yaml:9]: does this also need to be upaded?

Comment (izahn) [reply recipe/migrations/boost1780.yaml:9]: I don't know, this is my first time submitting a migration. Does it?

Comment (hmaarrfk) [reply recipe/migrations/boost1780.yaml:9]: i think it might be helpful. maybe. these "fixes" always confuse me

Comment (hmaarrfk) [reply recipe/migrations/boost1780.yaml:9]:
```
python -c "import time; print(f'{time.time():.0f}')"
```

Comment (izahn) [reply recipe/migrations/boost1780.yaml]: OK updated in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3403/commits/427c7752d071e66128e960501a1b4a4e7115101e

File: recipe/migrations/boost1780.yaml [hunk 2]
```diff
@@ -1,7 +1,9 @@
 __migrator:
   build_number: 1
   kind: version
-  migration_number: 1
+  migration_number: 2
 boost:
 - 1.78.0
-migrator_ts: 1649717600.9611828
+boost_cpp
```
Comment (izahn) [inline recipe/migrations/boost1780.yaml:7]: Oh hell there should be a colon at the end of this line right?

**Pull request merged**